enhancements/service-catalog/service-catalog-and-broker-removal.md

```diff
@@ -0,0 +1,211 @@
+In OpenShift 4.1, we announced the deprecation of the Service Catalog, the
+Ansible Service Broker, and the Template Service Broker. In 4.4, we plan to
+remove them from productization. They will no longer be installed in a new
+cluster. They will also block cluster upgrades until they are disabled in older
+clusters.
+
+## Motivation
+
+This is the next step in the removal of the Service Catalog and Brokers.
+
+* Deprecated APIs in OpenShift 4.1 [release
+  notes](https://docs.openshift.com/container-platform/4.1/release_notes/ocp-4-1-release-notes.html).
+* In OpenShift 4.2, a more detailed annoucement of their deprecation was
+  released: [4.2 release notes](https://docs.openshift.com/container-platform/4.2/release_notes/ocp-4-2-release-notes.html#ocp-41-deprecated-features)
+* [Support for templates](https://docs.openshift.com/container-platform/4.2/release_notes/ocp-4-2-release-notes.html#ocp-4-2-general-web-console-updates) without using the Template Service Broker and Service
+  Catalog added to OpenShift 4.2.
+* Alerts were added to the Service Catalog operators:
+  * https://github.com/openshift/cluster-svcat-apiserver-operator/blob/master/manifests/0000_90_cluster-svcat-apiserver-operator_02-operator-servicemonitor.yaml#L28-L45
+  * https://github.com/openshift/cluster-svcat-controller-manager-operator/blob/master/manifests/0000_90_cluster-svcat-controller-manager-operator_02_servicemonitor.yaml#L27-L44
+* Alerts were added to the Ansible Service Broker operator:
+  * https://github.com/openshift/ansible-service-broker/blob/master/operator/roles/ansible-service-broker/templates/broker.service-monitor.yaml.j2
+* Alerts were added to the Template Service Broker operator:
+  * https://github.com/openshift/template-service-broker-operator/blob/master/roles/template-service-broker/templates/tsb.prometheus-alert.yaml
+
+### Goals
+
+1. Implement a mechanism that ensures the Service Catalog and Service Brokers
+   are not present after an upgrade to OpenShift 4.4. (ENG)
+1. Prevent users from upgrading to OpenShift 4.4 if the Service Catalog or the
+   Service Brokers are in use. (ENG)
+1. Remove the Service Catalog from the OpenShift 4.4 release payload (ART)
+   * Will have to keep the Service Catalog operators if we choose to repurpose
+     them.
+1. Remove the Ansible Service Broker from the OpenShift 4.4 release payload (ART)
+1. Remove the Template Service Broker from the OpenShift 4.4 release payload (ART)
+1. Remove the Template Service Broker from productization (ART)
+1. Remove the Ansible Service Broker from productization (ART)
+1. Remove the Template Service Broker Operator from productization (ART)
+1. Remove the Ansible Service Broker Operator from productization (ART)
+1. Remove the Service Catalog from productization (ART)
+
+### Non-Goals
+
+N/A
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+Implement a mechanism that ensures the Service Catalog and Service Brokers are
+not present after an upgrade to OpenShift 4.4.
```

Comment:
Option 1 (block the upgrade) and option 2 (automatically remove) have some pretty significant differences from a UX standpoint. Is there a preference about which approach we should take? 

It seems like option 1 would be vastly preferable to force the decision on the cluster admin.

Is option 2 something we would seriously consider?